### ISSUE TITLE
Restore : Not pulling in newer nugets for Fixed dependencies

### ISSUE THREAD

**Issue by KevM** [FubuMvcArchive/ripple#154]
Upgraded one of our nugets and was revving downstream projects. 

``` xml
<Dependency Name="Dovetail.SDK" Version="3.2.10.27" Mode="Fixed" />
```
- Updated a nuget via ripple update --force on a fixed dependency. 

``` xml
<Dependency Name="Dovetail.SDK" Version="3.3.0.23" Mode="Fixed" />
```

It changed the ripple.config to have the new version and updated references and everything just fine as there was a .Net framework change from net20 to net40 which changed the hint paths.
- Committed the updated ripple.config and .csproj files.
- Other developers pulling the code and doing a rake (ripple restore) are not seeing the nuget get updated 

Annoyingly a side-effect of the old nuget getting used is the references all get rewritten to point at the old hint path (net20) which is how people noticed it quickly.

Workaround: **ripple restore --force**

**Comment by jeremydmiller** [comment]
It's not a new issue, but it's also not getting fixed w/o significant effort.  @jmarnold and I have tossed around the idea several times of making the "ripple:restore" task in fuburake be "ripple restore --force" or "ripple fix" but have never tried that.

**Comment by KevM** [comment]
Trying to understand why this is tricky. Is it that ripple restore doesn't know what version of the package is currently on disk or being referenced?  It seems the necessary metadata could be divined from or stored along side the exploded nuget.

**Comment by jmarnold** [comment]
It's not that we don't know, it's that often times we work with local
versions of the nugets. So you don't always want a restore to replace to
stomp over them when you run rake.

**Comment by KevM** [comment]
Ah, so isn't that the idea behind setting the dependency to fixed or float? Maybe a new type of dependency to communicate to Ripple how you want it to behave when deps versions change? :lock:

**Comment by jmarnold** [comment]
This is fixed in the 2.2 branch. It's not the final fix -- I want to introduce a more concrete "minimum version" concept but 2.2 honors the value supplied.
